core/src/tx.rs

```diff
@@ -116,25 +116,30 @@ impl TransactionValidator {
 
         if !is_genesis {
             debug!("Validating transaction: {:?}", tx);
-
-            let AcceptedTransaction {
-                payload,
-                signatures,
-            } = tx.clone();
-            let signatures = signatures.into_iter().collect();
-
-            let signed_tx = SignedTransaction {
-                payload,
-                signatures,
-            };
-            Self::validate_with_runtime_validators(
-                &signed_tx.payload.account_id.clone(),
-                signed_tx,
-                wsv,
-            )?;
+            Self::validate_with_runtime_validator(account_id, tx.clone(), wsv)?;
```

Comment:
why don't we validate genesis transactions?

Reply:
> Because genesis user is like an admin. That was done before me, but I think it makes sense
 
makes sense, but the question is why do we validate genesis instruction later in this function? Either genesis should be validated or it shouldn't?

@Erigara I remember you working on #2456 to disable limit checking for genesis. What is your opinion here?

Reply:
> but the question is why do we validate genesis instruction later in this function? Either genesis should be validated or it shouldn't?

The thing is that it's not a validation anymore, it's verification. And it requires to execute instructions to update `wsv`. That's what call-side expects

Reply:
> call it validation or verification. The point is the same,

**Language pedantry**: Validation is a side-effect free operation. Verification is not. More to the point Validation implies that something satisfies syntactic constraints, while verification also implies semantic meaning. So from the perspective of permissions, whether an instruction would fail to apply is semantic, as such a thing that checks both if the user is allowed to execute an instruction **and** if the instruction itself can be applied to the current WSV is a verifier _as a subset_ of validator. 

> it's not updating wsv, it's updating a clone of wsv. Transactions/instructions are executed/applied to live wsv in the next step

This is true at the moment. 

This isn't necessarily the case with patch-based WSV, or if we reverse instructions if they fail to apply. 

> execute these instructions on wsv, but panic here if even one transaction is invalid

I agree that this is the way to go.

Reply:
> Language pedantry: Validation is a side-effect free operation. Verification is not. More to the point Validation implies that something satisfies syntactic constraints, while verification also implies semantic meaning. So from the perspective of permissions, whether an instruction would fail to apply is semantic, as such a thing that checks both if the user is allowed to execute an instruction and if the instruction itself can be applied to the current WSV is a verifier as a subset of validator.

I was told of this suggestion made by you. Your point is valid (pun intended), this is why I opened #3368

Reply:
I think that panicking on genesis behaviour is already hapenning on Sumeragi side. What I was doing here is just updating old approach with genesis validation (verification). The main issue is about Monolithic Validator so, I think, let's move it to another issue

Reply:
Created #3382